### PR TITLE
[libos] Add support for address space layout randomization

### PR DESCRIPTION
**Pull request by jessehui**

### CODE REVIEW

File: src/libos/src/vm/process_vm.rs [hunk 1]
```diff
@@ -72,10 +72,12 @@ impl<'a, 'b> ProcessVMBuilder<'a, 'b> {
...
             .collect();
 
-        // Make heap and stack 16-byte aligned
+        // Make heap and stack page-aligned for simplicity
+        // Heap and stack can be 16-byte aligned, but with address space layout randomization, this can make many
+        // assertion failure.
         let other_layouts = vec![
-            VMLayout::new(heap_size, 16)?,
-            VMLayout::new(stack_size, 16)?,
+            VMLayout::new(heap_size, PAGE_SIZE)?,
```
Comment (guzongmin) [inline src/libos/src/vm/process_vm.rs:79]: Is it should be VMLayout::new(heap_size + PAGE_SIZE, 1)?,

Comment (jessehui) [reply src/libos/src/vm/process_vm.rs:79]: This layout is used for future allocation. Only required size and alignment are needed.

**Review by jessehui — COMMENTED** [review]
Thank you for your review.

**Review by guzongmin — APPROVED** [review]
LGTM

**Pull request merged**